### PR TITLE
Issue #5677: Refactor PackageObjectFactoryTest.testCreateModuleWithNonExistName

### PR DESCRIPTION
**Pull request by surajumang**
#5677

### CODE REVIEW

**Comment by surajumang** [comment]
Please let me know how should I proceed for PMD suppressions. I read about some ways to Suppress warnings [here](https://pmd.github.io/pmd-6.1.0/pmd_userdocs_suppressing_warnings.html) but I am not sure how should I start. @rnveach  was referring to use regex for this purpose. It would be very helpful if you be a little specific.

**Comment by rnveach** [comment]
@surajumang What warnings? CI is passing.

**Comment by surajumang** [comment]
@romani  @rnveach the issue #5677 also ask to Optimize PMD suppressions that is discussed in #5637 . I am talking about that. I haven't done anything in that regard.

**Comment by rnveach** [comment]
@surajumang All you need to do is remove the suppressions. `verify test` should still pass.
https://github.com/checkstyle/checkstyle/blob/master/config/pmd-test.xml#L90-L91
https://github.com/checkstyle/checkstyle/blob/master/config/pmd-test.xml#L102

File: config/pmd-test.xml [hunk 1]
```diff
@@ -99,9 +97,7 @@
               or @Image='SuppressWithNearbyCommentFilterTest' or @Image='SuppressionCommentFilterTest'
               or @Image='SuppressionFilterTest' or @Image='SuppressionXpathFilterTest'
               or @Image='SuppressWithPlainTextCommentFilterTest' or @Image='XpathFilterTest']//PrimarySuffix[@Image='verify']
-            | //ClassOrInterfaceDeclaration[@Image='PackageObjectFactoryTest']//MethodDeclarator[@Image='testCreateModuleWithNonExistName']
-            | //ClassOrInterfaceDeclaration[@Image='XdocsPagesTest']//MethodDeclarator[@Image='testAllChecksPresentOnAvailableChecksPage']
-            "/>
```
Comment (rnveach) [inline config/pmd-test.xml]: please restore this, `"/>` being alone on the next line, separate from the other suppressions.

**Comment by surajumang** [comment]
@rnveach The appveyor build is failing. I am unable to comprehend the Error log. Please suggest changes if any.

**Comment by rnveach** [comment]
Re-running it seems to have fixed it. I don't know what the error was about.

>  Exception calling "ExtractToDirectory" with "2" argument(s): "End of Central Directory record could not be found."
At line:7 char:3
>`+   [System.IO.Compression.ZipFile]::ExtractToDirectory("C:\maven-bin.z ...`
>`+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~`
>`    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException`
>`    + FullyQualifiedErrorId : InvalidDataException`
 
Command executed with exception: Exception calling "ExtractToDirectory" with "2" argument(s): "End of Central Directory record could not be found."

**Pull request merged**